### PR TITLE
feat(cli): add --output={text,json} to version cmd

### PR DESCRIPTION
**Pull request by johnstcn**
```
$ build/coder-slim_$(./scripts/version.sh)_linux_amd64 version --output=text
Coder v0.21.3-devel+b099e5c58 Wed Apr  5 12:03:29 UTC 2023
https://github.com/coder/coder/commit/b099e5c58c9f8dbb2fbbded3d559da7923cab3f8

Slim build of Coder, does not support the  server  subcommand.
$ build/coder-slim_$(./scripts/version.sh)_linux_amd64 version --output=json
{
  "version": "v0.21.3-devel+b099e5c58",
  "build_time": "2023-04-05T12:03:29Z",
  "external_url": "https://github.com/coder/coder/commit/b099e5c58c9f8dbb2fbbded3d559da7923cab3f8",
  "slim": true,
  "agpl": false
}
```

### CODE REVIEW

**Review by mafredri — APPROVED** [review]

File: cli/version.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+	"github.com/coder/coder/buildinfo"
+	"github.com/coder/coder/cli/clibase"
+	"github.com/coder/coder/cli/cliui"
+)
+
+// version prints the coder version
+func (*RootCmd) version() *clibase.Cmd {
+	handleHuman := func(inv *clibase.Invocation) error {
+		var str strings.Builder
```
Comment (mtojek) [inline cli/version.go:17]: nit: do you need a string builder if there is no way to break execution in the middle of processing?

**Review by mtojek — APPROVED** [review]
nit-picks can be ignored

**Comment by mafredri** [comment]
Hmm, should we call the flag output/format to align with table formatter and potentially support other formats in the future?

**Comment by johnstcn** [comment]
> Hmm, should we call the flag output/format to align with table formatter and potentially support other formats in the future?

Good point, we should have a consistent naming (although the current format is decidedly un-tabular)

**Comment by mafredri** [comment]
> Good point, we should have a consistent naming (although the current format is decidedly un-tabular)

Yeah, using a clibase enum with only `json` seems sufficient here. Only allowing `-{o,-output} json`. But users being able to rely on the `--output` flag seems like a good idea.

File: cli/version_test.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+			t.Parallel()
+			ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
+			t.Cleanup(cancel)
+			inv, _ := clitest.New(t, tt.Args...)
+			buf := new(bytes.Buffer)
+			inv.Stdout = buf
+			err := inv.WithContext(ctx).Run()
+			require.NoError(t, err)
+			actual := buf.String()
+			actual = strings.Replace(actual, "\r\n", "\n", -1)
```
[the remark this answers is not in the document]
Comment (johnstcn) [reply cli/version_test.go:62]: > > Good point, we should have a consistent naming (although the current format is decidedly un-tabular)
> 
> Yeah, using a clibase enum with only `json` seems sufficient here, only allowing `-{o,-output} json` (or maybe json and text, default being text). But users being able to rely on the `--output` flag seems like a good idea.

Turns out I needed to add a `textFormat` option to `cil/cliui` which wasn't too bad. It also forced me to somewhat abstract where we get the version info from, which isn't a bad thing. I haven't figured out how to plumb random version info into the unit tests yet, but that's a good follow-on PR.

**Comment by johnstcn** [comment]
> > Good point, we should have a consistent naming (although the current format is decidedly un-tabular)
> 
> Yeah, using a clibase enum with only `json` seems sufficient here, only allowing `-{o,-output} json` (or maybe json and text, default being text). But users being able to rely on the `--output` flag seems like a good idea.

Turns out I needed to add a `textFormat` option to `cli/cliui` which wasn't too bad. It also forced me to somewhat abstract where we get the version info from, which isn't a bad thing. I haven't figured out how to plumb random version info into the unit tests yet, but that's a good follow-on PR.

**Pull request merged**